core/network/warp/sync.cpp

```diff
@@ -115,6 +118,36 @@ namespace kagome::network {
     }
   }
 
+  inline auto guessSet(const GrandpaJustification &j) {
+    consensus::grandpa::AuthoritySetId set = 0;
+    static crypto::Ed25519ProviderImpl ed25519{nullptr};
+    auto &vote = j.items.at(0);
+    while (true) {
+      auto m = scale::encode(vote.message, j.round_number, set).value();
+      auto ok = ed25519.verify(vote.signature, m, vote.id);
+      if (ok and ok.value()) {
+        break;
+      }
+      ++set;
+    }
```

Comment:
Malicious peer can do DoS... 
1. needed reasonable limit of `set` value.
2. failure on encoding or verifying should break loop immediately with error (does not matter to continue in this case).

Reply:
Yes, new protocols should learn from this mistake by storing some redundant information like set id.

Added limit of block number (new set each block).